bindings/java/src/main/java/org/ucx/jucx/ucp/UcpEndpoint.java

```diff
@@ -58,6 +58,59 @@ public UcxRequest getNonBlocking(long remoteAddress, UcpRemoteKey remoteKey,
             dst, callback);
     }
 
+    /**
+     * Non-blocking tagged-send operations
+     * This routine sends a messages that is described by the local buffer {@code sendBuffer},
+     * to the destination endpoint. Each message is associated with a {@code tag} value
+     * that is used for message matching on the
+     * {@link UcpWorker#recvNonBlocking(ByteBuffer, long, UcxCallback)}
+     * "receiver".  The routine is non-blocking and therefore returns immediately,
+     * however the actual send operation may be delayed.
+     * The send operation is considered completed when  it is safe to reuse the source
+     * {@code data} buffer. {@code callback} is invoked on completion of this operation.
+     */
+    public UcxRequest sendNonBlocking(ByteBuffer sendBuffer, long tag, UcxCallback callback) {
+        if (!sendBuffer.isDirect()) {
+            throw new UcxException("Send buffer must be direct.");
+        }
+        if (callback == null) {
+            callback = new UcxCallback();
+        }
+        return sendNonBlockingNative(getNativeId(), sendBuffer, tag, callback);
+    }
+
+    /**
+     * Non blocking send operation. Invokes
+     * {@link UcpEndpoint#sendNonBlocking(ByteBuffer, long, UcxCallback)} with default 0 tag.
+     */
+    public UcxRequest sendNonBlocking(ByteBuffer sendBuffer, UcxCallback callback) {
+        return sendNonBlocking(sendBuffer, 0, callback);
+    }
+
+    /**
+     * Non-blocking remote memory put operation.
+     * This routine initiates a storage of contiguous block of data that is
+     * described by the local {@code data} buffer in the remote contiguous memory
+     * region described by {@code remoteAddress} address and the {@code remoteKey} "memory
+     * handle". The routine returns immediately and <strong>does</strong> not
+     * guarantee re-usability of the source {@code data} buffer.
+     * {@code callback} is invoked on completion of this operation.
+     */
+    public UcxRequest putNonBlocking(ByteBuffer src, long remoteAddress, UcpRemoteKey remoteKey,
+                                     UcxCallback callback) {
+        if (!src.isDirect()) {
```

Comment:
can we wrap these 2 checks to helper function: checkRemoteAccessParams(ByteBuffer src, UcpRemoteKey remoteKey)